File: Server/src/main/java/com/mancersoft/litevpnserver/transport/Packet.java
package com.mancersoft.litevpnserver.transport;

public class Packet {

    private Object mSource;
    private Object mDestination;
    private byte[] mData;
    private int mLength;

    public Object getSource() {
        return mSource;
    }

    public void setSource(Object source) {
        this.mSource = source;
    }

    public Object getDestination() {
        return mDestination;
    }

    public void setDestination(Object destination) {
        this.mDestination = destination;
    }

    public byte[] getData() {
        return mData;
    }

    public void setData(byte[] data) {
        this.mData = data;
    }

    public int getLength() {
        return mLength;
    }

    public void setLength(int length) {
        this.mLength = length;
    }
}

File: Server/src/main/kotlin/com/mancersoft/litevpnserver/transport/Packet.kt
package com.mancersoft.litevpnserver.transport

class Packet {
    var source: Any? = null
    val sourceId: String?
        get() = source?.toString()
    var destination: Any? = null
    val destId: String?
        get() = destination?.toString()
    lateinit var data: ByteArray
    var length = 0

    constructor()
    constructor(packet: Packet) {
        source = packet.source
        destination = packet.destination
        data = ByteArray(packet.length)
        System.arraycopy(packet.data, 0, data, 0, packet.length)
        length = packet.length
    }

}